### PR TITLE
Port allmydata.util.log to Python 3

### PR DESCRIPTION
**Pull request by itamarst**
Fixes https://tahoe-lafs.org/trac/tahoe-lafs/ticket/3365

### CODE REVIEW

File: src/allmydata/test/test_log.py [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+
+    def test_err(self):
+        """Logging with log.err() causes tests to fail."""
+        try:
+            raise SampleError("simple sample")
+        except:
+            f = Failure()
+        tahoe_log.err(format="intentional sample error",
+                      failure=f, level=tahoe_log.OPERATIONAL, umid="wO9UoQ")
+        self.flushLoggedErrors(SampleError)
```
Comment (exarkun) [inline src/allmydata/test/test_log.py:62]: I'd probably add an assertion here, that `flushLoggedErrors` returned a list with one element, and change the docstring to just say that `log.err` forwards errors to the Twisted log (and then it's up to trial to make the test fail, not our concern).

File: src/allmydata/test/test_log.py [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+
+    def test_default_facility(self):
```
Comment (exarkun) [inline src/allmydata/test/test_log.py:64]: This and the other test methods could use a quick docstring.

File: src/allmydata/test/test_log.py (hunk 1 above)
Comment (exarkun) [reply src/allmydata/test/test_log.py:62]: Oh.  I see.  This test was moved from the other module. :/  Where it was also testing nothing...  Well, the changes would still be good, I guess, and they're basically trivial.

File: src/allmydata/test/test_log.py (hunk 2 above)
Comment (itamarst) [reply src/allmydata/test/test_log.py:64]: I guess I will figure out what "facility" means, then :)

**Comment by itamarst** [comment]
Looks like some legit test failures.

**Comment by itamarst** [comment]
More test failures :(

**Pull request merged**